controller/iteration_blackbox_test.go

```diff
@@ -763,3 +765,208 @@ func assertChildIterationLinking(t *testing.T, target *app.Iteration) {
+					if idx == 1 {
+						fxt.Iterations[idx].MakeChildOf(*fxt.Iterations[0])
+					}
+					return nil
+				}))
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		iterationToDelete := fxt.IterationByName("first iteration")
+		test.DeleteIterationOK(t, svc.Context, svc, ctrl, iterationToDelete.ID)
+		_, err := rest.db.Iterations().Load(svc.Context, iterationToDelete.ID)
+		require.NotNil(t, err)
+		require.IsType(t, errors.NotFoundError{}, err, "error was %v", err)
+	})
+
+	rest.T().Run("success - delete iteration subtree", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB,
+			tf.Iterations(6,
+				tf.SetIterationNames("root", "child 1", "child 1.2", "child 1.2.3", "child 1.2.3.4", "child 2"),
+				func(fxt *tf.TestFixture, idx int) error {
+					i := fxt.Iterations[idx]
+					switch idx {
+					case 1:
+						i.MakeChildOf(*fxt.Iterations[0])
+					case 2:
+						i.MakeChildOf(*fxt.Iterations[1])
+					case 3:
+						i.MakeChildOf(*fxt.Iterations[2])
+					case 4:
+						i.MakeChildOf(*fxt.Iterations[3])
+					case 5:
+						i.MakeChildOf(*fxt.Iterations[0])
+					}
+					return nil
+				}))
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		iterationToDelete := fxt.IterationByName("child 1")
+		test.DeleteIterationOK(t, svc.Context, svc, ctrl, iterationToDelete.ID)
+		// make sure all nested iterations are deleted
+		deletedIterations := []*iteration.Iteration{
+			fxt.IterationByName("child 1"),
+			fxt.IterationByName("child 1.2"),
+			fxt.IterationByName("child 1.2.3"),
+			fxt.IterationByName("child 1.2.3.4"),
+		}
+		for _, i := range deletedIterations {
+			_, err := rest.db.Iterations().Load(svc.Context, i.ID)
+			require.NotNil(t, err)
+			require.IsType(t, errors.NotFoundError{}, err, "error was %v", err)
+		}
+		// make sure other iterations are not touched
+		iterationsShouldPresent := []*iteration.Iteration{
+			fxt.IterationByName("root"),
+			fxt.IterationByName("child 2"),
+		}
+		for _, i := range iterationsShouldPresent {
+			_, err := rest.db.Iterations().Load(svc.Context, i.ID)
+			require.Nil(t, err)
+		}
+	})
+
+	rest.T().Run("forbidden - other user can not delete iteration", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB,
+			tf.Identities(2, tf.SetIdentityUsernames("space owner", "other user")),
+			tf.Iterations(1))
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.IdentityByUsername("other user"))
+		test.DeleteIterationForbidden(t, svc.Context, svc, ctrl, fxt.Iterations[0].ID)
+	})
+
+	rest.T().Run("success - space owner can delete iteration", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.Iterations(2, func(fxt *tf.TestFixture, idx int) error {
+			if idx == 1 {
+				fxt.Iterations[idx].MakeChildOf(*fxt.Iterations[0])
+			}
+			return nil
+		}))
+		iterationToDelete := fxt.Iterations[1]                             // non-root iteration
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0]) // get the sapce owner
```

Comment:
minor typo: "space owner"

Reply:
my bad, fixed in https://github.com/fabric8-services/fabric8-wit/pull/1683/commits/21a922cabd1ddca24311e954ec69455efd52b824